python/ray/serve/_private/replica.py

```diff
@@ -218,78 +213,113 @@ async def handle_request(
             pickled_request_metadata: bytes,
             *request_args,
             **request_kwargs,
-        ):
-            query = Query(
-                request_args,
-                request_kwargs,
-                pickle.loads(pickled_request_metadata),
+        ) -> Tuple[bytes, Any]:
+
+            request_metadata = pickle.loads(pickled_request_metadata)
+            if request_metadata.is_http_request:
+                assert len(request_args) == 1
```

Comment:
Non-blocker: Would be nice to add a message here to note what this is checking. HTTP request requires exactly one request argument or something similar :)